features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/modules/group.js

```diff
@@ -0,0 +1,193 @@
+        //URL: DELETE https://localhost:9443/devicecloud/group_manager/group/id/{groupId}/role
+        endPoint = deviceCloudService + "/group/id/" + groupId + "/role/" + role;
+        return del(endPoint, {}, "json");
+    };
+
+    groupModule.getGroupRoles = function (groupId) {
+        //URL: GET https://localhost:9443/devicecloud/group_manager/group/id/{groupId}/role/all
+        endPoint = deviceCloudService + "/group/id/" + groupId + "/role/all";
+        data = {"username": user.username};
+        return get(endPoint, data, "json");
+    };
+
+    groupModule.getUserRoles = function (groupId, userId) {
+        //URL: GET https://localhost:9443/devicecloud/group_manager/group/id/{groupId}/{user}/role/all
+        endPoint = deviceCloudService + "/group/id/" + groupId + "/" + userId + "/role/all";
+        data = {"username": user.username};
+        return get(endPoint, data, "json");
+    };
+
+    groupModule.getRoleMapping = function (groupId, userId) {
+        var allRoles = groupModule.getGroupRoles(groupId).data;
+        var userRolesObj = groupModule.getUserRoles(groupId, userId);
+        var userRoles = userRolesObj.data;
+        var roleMap = [];
+        for (var role in allRoles) {
+            var objRole = {"role": allRoles[role], "assigned": false};
+            for (var usrRole in userRoles) {
+                if (allRoles[role] == userRoles[usrRole]) {
+                    objRole.assigned = true;
+                    break;
+                }
+            }
+            roleMap.push(objRole);
+        }
+        var result = {};
+        result.data = roleMap;
+        result.xhr = userRolesObj.xhr;
+        return result;
+    };
+
+    groupModule.setRoleMapping = function (groupId, userId, roleMap) {
+        var result;
+        for (var role in roleMap) {
+            if (roleMap[role].assigned == true) {
+                result = groupModule.shareGroup(groupId,userId,roleMap[role].role);
+            } else {
+                result = groupModule.unshareGroup(groupId,userId,roleMap[role].role);
+            }
+        }
+        return result;
+    };
+
+    groupModule.getUsers = function (groupId) {
+        //URL: GET https://localhost:9443/devicecloud/group_manager/group/id/{groupId}/user/all
+        endPoint = deviceCloudService + "/group/id/" + groupId + "/user/all";
+        data = {"username": user.username};
+        return get(endPoint, data, "json");
+    };
+
+    groupModule.getDevices = function (groupId) {
+        var result = groupModule.getGroup(groupId);
+        var group = result.data;
+        if (group) {
+            //URL: GET https://localhost:9443/devicecloud/group_manager/group/id/{groupId}/device/all
+            endPoint = deviceCloudService + "/group/id/" + groupId + "/device/all";
+            data = {"username": user.username};
+            result = get(endPoint, data, "json");
+            var devices = result.data;
+            for (var d in devices){
```

Comment:
Acknowledge, Already fixed in moved code with commit https://github.com/wso2/carbon-device-mgt/commit/e537e2d841ca3c7946cd34e9da96f261e0ac94e1